src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Operators.cs

```diff
@@ -79,9 +80,105 @@ internal sealed partial class ExpressionEval {
                 return null;
             }
 
+            var op = binop.Operator;
+
+            var left = GetValueFromExpression(binop.Left) ?? UnknownType;
+            var right = GetValueFromExpression(binop.Right) ?? UnknownType;
+
+            if (left.IsUnknown() && right.IsUnknown()) {
+                // Fast path for when nothing below will give any results.
+                if (op.IsComparison()) {
+                    return Interpreter.GetBuiltinType(BuiltinTypeId.Bool);
+                }
+
+                return UnknownType;
+            }
+
+            var leftType = left.GetPythonType();
+            var rightType = right.GetPythonType();
+
+            if (leftType == null) {
```

Comment:
Probably never null since you do `?? UnknownType`

Reply:
I think it could still be null if I were trying to compare a module to something else (or some other case the GetPythonType extension isn't handling). I'll test it and see.

Reply:
I tested things, and these don't appear to ever be null.